satellite/gracefulexit/endpoint.go

```diff
@@ -922,3 +777,97 @@ func (endpoint *Endpoint) updatePointer(ctx context.Context, originalPointer *pb
 
 	return nil
 }
+
+func (endpoint *Endpoint) checkExitStatus(ctx context.Context, nodeID storj.NodeID) (*pb.SatelliteMessage, error) {
+	exitStatus, err := endpoint.overlaydb.GetExitStatus(ctx, nodeID)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	if exitStatus.ExitFinishedAt != nil {
+		// TODO maybe we should store the reason in the DB so we know how it originally failed.
+		return endpoint.getFinishedMessage(ctx, endpoint.signer, nodeID, *exitStatus.ExitFinishedAt, exitStatus.ExitSuccess, -1)
+	}
+
+	if exitStatus.ExitInitiatedAt == nil {
+		request := &overlay.ExitStatusRequest{NodeID: nodeID, ExitInitiatedAt: time.Now().UTC()}
+		node, err := endpoint.overlaydb.UpdateExitStatus(ctx, request)
+		if err != nil {
+			return nil, Error.Wrap(err)
+		}
+		err = endpoint.db.IncrementProgress(ctx, nodeID, 0, 0, 0)
+		if err != nil {
+			return nil, Error.Wrap(err)
+		}
+
+		// graceful exit initiation metrics
+		age := time.Now().UTC().Sub(node.CreatedAt.UTC())
+		mon.FloatVal("graceful_exit_init_node_age_seconds").Observe(age.Seconds())
```

Comment:
nit - maybe since we are moving everything around anyway, we should go ahead and add `//locked` to these monkit values so that they are added to `monkit.lock`. We can also do it separately, so don't worry too much about it.